### DISCUSSION TITLE
Can I contribute?

### DISCUSSION THREAD

**Discussion by JimWilkerson** [DSheirer/sdrtrunk#1729] [General]
I am a career Java coder and use sdrtrunk extensively.  I've been building the master branch and stepping through it.

If you have a to-do list I would love to chip in.

**Comment by DSheirer** [comment]
@JimWilkerson yes, you're more than welcome to contribute. Do you have any areas/technologies where you'd prefer to work, or do you want to work smaller issues/bug fixes until you get acquainted with the codebase?

Some focus areas for me are (in no particular order):
1. Audio call management sub-system redesign.
2. Create an API/endpoint to access calls information
3. Create an API/endpoint for control/configuration of playlists.
4. Add embedded HTTP server with Javascript or Typescript hosting frontend over the calls API (2) and eventually (3)
5. Port all Swing UI components to JavaFX (I'm working this little by little)
6. JavaFX-based spectrum/waterfall display replacement
7. Incorporate GPU processing to offload DSP workloads (FFT, filtering, demodulation, etc).  I've looked at Aparapi and TornadoVM and I'm wondering if maybe there's another option using a custom Foreign Function/Foreign Memory wrapper around OpenCL, since OpenCL has common support.
8. Runtime optimization of FFT processing (integer/short samples vs floating point) for SBCs like Raspberry Pi 4/5.  This probably ties into 7
9. Create an NXDN decoder.
10. Create a Motorola Type II (3600-baud) decoder.
11. Additional tuner support (Ettus, LimeSDR, xtrx, etc)

**Comment by JimWilkerson** [comment]
I am happy to work bugs and fixes and get my cops to start.  I would say my
areas of expertise correspond best with the REST and HTTP work above,  2-4.
That's what I do at work.  The radio stuff is a hobby.

**Comment by DSheirer** [comment]
Would you like to attempt this one: https://github.com/DSheirer/sdrtrunk/issues/1736
